docs/DeveloperGuide.md

```diff
@@ -1,34 +1,522 @@
+
+Additionally, the add module command extends the `Command` class and overrides its `execute()` command. An external class, `ModuleValidator` is called upon to validate the various parameters that the user has entered, as to only allow valid modules to be added to the user.
+
+Given below is an example usage scenario and how add module command behaves at each step.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_initialState.png" alt="Initial state diagram for AddModuleCommand"/>
+</div>
+**Step 1** : The user calls the add module command from the `AcademicPlannerParser`, which will initialise a 
+`AddModuleCommand`. `AddModuleCommand`'s constructor takes in parameters of `ModuleLoader`, `Person`,`Ui`, 
+and `String`. Below is a table of what each parameter corresponds to in the state diagram of the program.
+
+|Parameter<br />(Class Name)|Corresponds to<br />(Function of Class)|Referred to as<br />(Variable Name)|
+|:---:|:---:|:---:|
+|`ModuleLoader`| Class representing all modules offered by NUS | `allModules` |
+|`Person`| Class representing current user's information | `currentPerson`|
+|`Ui`| Class representing java's default scanner class | `in`|
+|`String` | Class representing the module code to be added | `moduleCode`|
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_state2.png" alt="State diagram for AddModuleCommand Step 2"/>
+</div>
+**Step 2** : `execute()` is called from the instance of `AddModuleCommand`. It can throw `AcademicException` 
+or `IOException`. `FileHandler` and `Logger` classes from the _java API_ are instantiated to handle logging for the remainder of the `execute()` method. 
+
+**Step 3** : `in` then reads in the next two lines of input, which is the user's input for the desired semester for the `moduleCode` and `moduleCode`'s grades.
+
+**Step 4** : `validateInputs()` is called to validate the user entered data against `allModules`.
+
+**Step 5** : `AddUtils` is called upon to return module credit for `moduleCode` by `getModuleCreditForModule()`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_state6.png" alt="State diagram for AddModuleCommand Step 6"/>
+</div>
+
+**Step 6** :  `AddUtils` is called upon again to add the module's data to the user, by instantiating a new
+`PartialModule` and storing it in both `userModuleList` and `userModuleMap` via `Person`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_finalState.png" alt="Final state diagram for AddModuleCommand"/>
+</div>
+
+**Step 7** : `FileHandler`, `Logger`, `PartialModule`, `ModuleValidator`, `AddUtils` and `AddModuleCommand` are terminated.
+
+The following sequence diagram shows how the `AddModuleCommand` works:
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_sequence.png" alt="Sequence diagram for AddModuleCommand"/>
+</div>
```

Comment:
This sequence diagram is very clear and the previous diagrams explained each step clearly! Well done.